File: backend/SharedLib/DTOs/Mood/MoodEntryDto.cs
namespace SharedLib.DTOs.Mood
{
    public record MoodEntryDto(Guid Id, DateTime Day, string MoodTime, string MoodTimeRange, string MoodLevel, int MoodScore, string Note, DateTime CreatedAt, Guid UserId)
    {
        public string DayStr => Day.ToString("yyyy-MM-dd");
        public string CreatedAtStr => CreatedAt.ToShortDateString();
    }
}

File: frontend/devpulse-ui/src/app/core/models/mood-entry.dto.ts
import { BaseDto } from './base.dto';

export interface MoodEntryDto extends BaseDto {
  day: string; // ISO string or date string
  moodTime: string;
  moodTimeRange: string;
  moodLevel: string;
  moodScore: number;
  note: string;
  createdAt: string; // ISO string or date string
  userId: string; // Guid

  dayStr: string; // computed on backend
  createdAtStr: string; // computed on backend
}
